q2_longitudinal/_utilities.py

```diff
@@ -673,6 +673,174 @@ def _nmit(table, sample_md, individual_id_column, corr_method="kendall",
+            data=output.iloc[:, 2].values, name='Distance')
+
+
+def _first_distances_and_distance_to_baseline(metadata, state_column,
+                                              individual_id_column,
+                                              distance_matrix,
+                                              replicate_handling='error',
+                                              baseline=None):
+    distance_matrix = distance_matrix.to_data_frame()
+
+    metadata = metadata[[state_column, individual_id_column]]
+    metadata = metadata.loc[distance_matrix.index]
+
+    # let's force states to be numeric
+    _validate_is_numeric_column(metadata, state_column)
+
+    # combine individual and state to vectorize checks
+    combo_column = _generate_column_name(metadata)
+    metadata[combo_column] = (metadata[state_column].astype(float).astype(str)
+                              + metadata[individual_id_column].astype(str))
+    if replicate_handling == 'drop':
+        duplicated = metadata.duplicated(subset=combo_column, keep=False)
+    else:
+        # this way of finding duplicates is relevant for "random" too
+        duplicated = metadata.duplicated(subset=combo_column)
+        if replicate_handling == 'error':
+            if duplicated.any():
+                def summarizer(group, id_column=None):
+                    out = group[id_column].iloc[0]
+                    out = '%s: %s' % (out, ', '.join(group[state_column]))
+                    return out
+                messages = metadata.groupby(
+                    individual_id_column
+                ).apply(summarizer, id_column=individual_id_column).tolist()
+                messages = "\n".join(messages)
+                raise ValueError('There are repeated states for the'
+                                 'following individuals.\n'
+                                 f'{messages}')
+
+    # depending on the strategy selected above filter the table
+    metadata = metadata[~duplicated].copy()
+
+    # if calculating static differences, validate baseline as a valid state
+    if baseline is not None:
+        # convert baseline to the column's type
+        baseline = metadata[state_column].dtype.type(baseline)
+
+        # check that baseline is present for all individuals
+        have_baseline = metadata.groupby(
+            individual_id_column).apply(
+                lambda group: baseline in group[state_column].values)
+        if not have_baseline.all():
+            missing = ', '.join([str(i) for i in
+                                 have_baseline[~have_baseline].index])
+            raise ValueError('baseline must be a valid state. The following '
+                             'individuals are missing a baseline'
+                             f' state value of "{baseline}": {missing}')
+
+        # use -np.inf to always make the baseline the first in the list
+        metadata.replace({baseline: -np.inf}, inplace=True)
```

Comment:
this will replace all matching values in the entire df, correct? instead of just the `state_column` column?

Reply:
this also relates to my question about `_vectorized_distance_to_baseline`... instead of doing a pass over the df to `replace` baseline values now, maybe this could just be done inside that function when grabbing the baseline row?